app/lib/katello/validators/environment_docker_repositories_validator.rb

```diff
@@ -0,0 +1,78 @@
+module Katello
+  module Validators
+    class EnvironmentDockerRepositoriesValidator < ActiveModel::Validator
+      def validate(environment)
+        return true if environment.registry_name_pattern.empty?
+
+        unless ContainerImageNameValidator.validate_name(Katello::Repository.safe_render_container_name(test_repository, environment.registry_name_pattern))
+          environment.errors.add(:registry_name_pattern, N_("Registry name pattern will result in invalid container image name of member repositories"))
+          return false
+        end
+
+        error_messages = EnvironmentDockerRepositoriesValidator.validate_repositories(environment.registry_name_pattern, environment.repositories.docker_type)
+        return true if error_messages.empty?
+
+        error_messages.each do |message|
+          environment.errors.add(:registry_name_pattern, message)
+        end
+        false
+      end
+
+      def self.validate_repositories(registry_name_pattern, repositories)
+        error_messages = []
+        names = []
+        repositories.each do |repository|
+          name = Katello::Repository.safe_render_container_name(repository, registry_name_pattern)
+
+          unless ContainerImageNameValidator.validate_name(name)
+            error_messages << N_("Registry name pattern results in invalid container image name of member repository '%{name}'") % {name: repository.name}
+            return error_messages
+          end
+          names << name
+        end
+
+        if names.length != names.uniq.length
+          error_messages << N_("Registry name pattern results in duplicate container image names")
+          return error_messages
+        end
+
+        Katello::Repository.where(content_type: Katello::Repository::DOCKER_TYPE).each do |repository|
+          if !repositories.include?(repository) && names.include?(repository.container_repository_name)
```

Comment:
@jlsherrill - Updated this with original intention: Check all repos except for the current ones being processed (they're checked above) for name clashes.

Really struggling to write sensible tests.